// reakit/lib/chunk-b0c0085a.js
'use strict';

var __chunk_1 = require('./chunk-4a9d70be.js');
var styledTools = require('styled-tools');
var styled$1 = require('./styled.js');
var use$1 = require('./use.js');
var React = require('react');
var PropTypes = require('prop-types');
var Box = require('./Box.js');
var __chunk_4 = require('./chunk-fa7792e6.js');
var __chunk_5 = require('./chunk-c40dcb9e.js');

function _templateObject() {
  var data = __chunk_1._taggedTemplateLiteralLoose(["\n  ", ";\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

var HiddenHideComponent = function HiddenHideComponent(_ref) {
  var onClick = _ref.onClick,
      props = __chunk_1._objectWithoutPropertiesLoose(_ref, ["onClick"]);

  return React.createElement(Box.default, __chunk_1._extends({
    onClick: __chunk_5.callAll(props.hide, onClick)
  }, props));
};

var HiddenHide = styled$1.default(__chunk_4.hoist(HiddenHideComponent, Box.default))(_templateObject(), styledTools.theme("HiddenHide")); // @ts-ignore

HiddenHide.propTypes = {
  hide: PropTypes.func.isRequired,
  onClick: PropTypes.func
};
var HiddenHide$1 = use$1.default(HiddenHide, "button");

function _templateObject$1() {
  var data = __chunk_1._taggedTemplateLiteralLoose(["\n  ", ";\n"]);

  _templateObject$1 = function _templateObject() {
    return data;
  };

  return data;
}

var HiddenShowComponent = function HiddenShowComponent(_ref) {
  var onClick = _ref.onClick,
      props = __chunk_1._objectWithoutPropertiesLoose(_ref, ["onClick"]);

  return React.createElement(Box.default, __chunk_1._extends({
    onClick: __chunk_5.callAll(props.show, onClick)
  }, props));
};

var HiddenShow = styled$1.default(__chunk_4.hoist(HiddenShowComponent, Box.default))(_templateObject$1(), styledTools.theme("HiddenShow")); // @ts-ignore

HiddenShow.propTypes = {
  show: PropTypes.func.isRequired,
  onClick: PropTypes.func
};
var HiddenShow$1 = use$1.default(HiddenShow, "button");

function _templateObject$2() {
  var data = __chunk_1._taggedTemplateLiteralLoose(["\n  ", ";\n"]);

  _templateObject$2 = function _templateObject() {
    return data;
  };

  return data;
}
var HiddenToggleComponent = React.forwardRef(function (_ref, ref) {
  var onClick = _ref.onClick,
      props = __chunk_1._objectWithoutPropertiesLoose(_ref, ["onClick"]);

  return React.createElement(Box.default, __chunk_1._extends({
    onClick: __chunk_5.callAll(props.toggle, onClick)
  }, props, {
    ref: ref
  }));
});
var HiddenToggle = styled$1.default(__chunk_4.hoist(HiddenToggleComponent, Box.default))(_templateObject$2(), styledTools.theme("HiddenToggle")); // @ts-ignore

HiddenToggle.propTypes = {
  toggle: PropTypes.func.isRequired,
  onClick: PropTypes.func
};
var HiddenToggle$1 = use$1.default(HiddenToggle, "button");

exports.HiddenHide = HiddenHide$1;
exports.HiddenShow = HiddenShow$1;
exports.HiddenToggle = HiddenToggle$1;
