// reakit/lib/Code.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var __chunk_1 = require('./chunk-4a9d70be.js');
var styledTools = require('styled-tools');
require('styled-components');
var styled$1 = require('./styled.js');
require('reuse');
var use$1 = require('./use.js');
var React = require('react');
var PropTypes = require('prop-types');
require('./chunk-1cbeb596.js');
require('./chunk-e7befa1f.js');
var Box = require('./Box.js');
require('@emotion/is-prop-valid');

function _templateObject() {
  var data = __chunk_1._taggedTemplateLiteralLoose(["\n  ", ";\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

var CodeComponent = function CodeComponent(props) {
  if (props.block) {
    return React.createElement(Box.default, __chunk_1._extends({
      use: "pre"
    }, props), React.createElement("code", {
      className: props.codeClassName
    }, props.children));
  }

  var className = props.codeClassName ? (props.className + " " + props.codeClassName).trim() : ("" + props.className).trim();
  return React.createElement(Box.default, __chunk_1._extends({
    use: "code"
  }, props, {
    className: className
  }));
};

var Code = styled$1.default(Box.default)(_templateObject(), styledTools.theme("Code")); // @ts-ignore

Code.propTypes = {
  block: PropTypes.bool,
  codeClassName: PropTypes.string
};
Code.defaultProps = {
  use: CodeComponent,
  opaque: true,
  palette: "background",
  tone: -2
};
var Code$1 = use$1.default(Code);

exports.default = Code$1;
