// reakit/lib/Grid.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var __chunk_1 = require('./chunk-4a9d70be.js');
var styledTools = require('styled-tools');
require('styled-components');
var styled$1 = require('./styled.js');
require('reuse');
var use$1 = require('./use.js');
require('react');
var PropTypes = require('prop-types');
require('./chunk-1cbeb596.js');
var __chunk_3 = require('./chunk-e7befa1f.js');
var Box = require('./Box.js');
require('@emotion/is-prop-valid');

function _templateObject() {
  var data = __chunk_1._taggedTemplateLiteralLoose(["\n  display: grid;\n  &&& {\n    ", "\n    ", "\n    ", "\n    ", "\n    ", "\n    ", "\n    ", "\n    ", "\n  }\n  ", ";\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}
var Grid = styled$1.default(Box.default)(_templateObject(), __chunk_3.bool("grid-auto-flow", ["row", "column", "dense"]), __chunk_3.value("grid-gap", "gap"), __chunk_3.value("grid-template", "template"), __chunk_3.value("grid-template-areas", "templateAreas"), __chunk_3.value("grid-template-columns", "templateColumns"), __chunk_3.value("grid-template-rows", "templateRows"), __chunk_3.value("grid-auto-columns", "autoColumns"), __chunk_3.value("grid-auto-rows", "autoRows"), styledTools.theme("Grid"));
var valueType = PropTypes.oneOfType([PropTypes.string, PropTypes.number]); // @ts-ignore

Grid.propTypes = {
  row: PropTypes.bool,
  column: PropTypes.bool,
  dense: PropTypes.bool,
  gap: valueType,
  template: valueType,
  templateAreas: valueType,
  templateColumns: valueType,
  templateRows: valueType,
  autoColumns: valueType,
  autoRows: valueType
};
var Grid$1 = use$1.default(Grid, "div");

function _templateObject$1() {
  var data = __chunk_1._taggedTemplateLiteralLoose(["\n  &&& {\n    ", ";\n    ", ";\n    ", ";\n    ", ";\n    ", ";\n    ", ";\n    ", ";\n  }\n  ", ";\n"]);

  _templateObject$1 = function _templateObject() {
    return data;
  };

  return data;
}
var GridItem = styled$1.default(Box.default)(_templateObject$1(), __chunk_3.value("grid-area", "area"), __chunk_3.value("grid-column", "column"), __chunk_3.value("grid-row", "row"), __chunk_3.value("grid-column-start", "columnStart"), __chunk_3.value("grid-column-end", "columnEnd"), __chunk_3.value("grid-row-start", "rowStart"), __chunk_3.value("grid-row-end", "rowEnd"), styledTools.theme("GridItem"));
var valueType$1 = PropTypes.oneOfType([PropTypes.string, PropTypes.number]); // @ts-ignore

GridItem.propTypes = {
  area: valueType$1,
  column: valueType$1,
  row: valueType$1,
  columnStart: valueType$1,
  columnEnd: valueType$1,
  rowStart: valueType$1,
  rowEnd: valueType$1
};
var GridItem$1 = use$1.default(GridItem, "div");

var index = Object.assign(Grid$1, {
  Item: GridItem$1
});

exports.default = index;
