// reakit/lib/Overlay.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var __chunk_1 = require('./chunk-4a9d70be.js');
var styledTools = require('styled-tools');
require('styled-components');
var styled$1 = require('./styled.js');
require('reuse');
var use$1 = require('./use.js');
require('react');
require('prop-types');
require('./chunk-1cbeb596.js');
require('./chunk-e7befa1f.js');
require('./Box.js');
require('@emotion/is-prop-valid');
require('react-dom');
require('./chunk-fa7792e6.js');
require('./chunk-c40dcb9e.js');
var Hidden = require('./Hidden.js');
require('constate');
require('./chunk-9a6ca4f6.js');
require('./chunk-b0c0085a.js');
var __chunk_8 = require('./chunk-a45b3f05.js');

function _templateObject() {
  var data = __chunk_1._taggedTemplateLiteralLoose(["\n  position: fixed;\n  z-index: 19900410;\n  left: 50%;\n  top: 50%;\n  ", ";\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}
var Overlay = styled$1.default(Hidden.default)(_templateObject(), styledTools.theme("Overlay"));
Overlay.defaultProps = {
  role: "dialog",
  "aria-modal": true,
  hideOnEsc: true,
  translateX: "-50%",
  translateY: "-50%",
  defaultSlide: "top",
  opaque: true,
  palette: "background",
  tone: -1
};
var Overlay$1 = use$1.default(Overlay, "div");

var Overlay$2 = Object.assign(Overlay$1, {
  Container: __chunk_8.OverlayContainer,
  Toggle: __chunk_8.OverlayToggle,
  Show: __chunk_8.OverlayShow,
  Hide: __chunk_8.OverlayHide
});

exports.default = Overlay$2;
