// reakit/lib/Popover.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

function _interopDefault (ex) { return (ex && (typeof ex === 'object') && 'default' in ex) ? ex['default'] : ex; }

var __chunk_1 = require('./chunk-4a9d70be.js');
var styledTools = require('styled-tools');
require('styled-components');
var styled$1 = require('./styled.js');
require('reuse');
var use$1 = require('./use.js');
var React = require('react');
var PropTypes = require('prop-types');
require('./chunk-1cbeb596.js');
require('./chunk-e7befa1f.js');
require('./Box.js');
require('@emotion/is-prop-valid');
var ReactDOM = require('react-dom');
var __chunk_4 = require('./chunk-fa7792e6.js');
var __chunk_5 = require('./chunk-c40dcb9e.js');
var Hidden = require('./Hidden.js');
require('constate');
var __chunk_6 = require('./chunk-9a6ca4f6.js');
var __chunk_7 = require('./chunk-b0c0085a.js');
var s = _interopDefault(require('styled-selector'));
var Popper = _interopDefault(require('popper.js'));
var __chunk_9 = require('./chunk-dbf1c71e.js');

function _templateObject() {
  var data = __chunk_1._taggedTemplateLiteralLoose(["\n  position: absolute;\n  top: 0;\n  left: 0;\n  user-select: auto;\n  cursor: auto;\n  z-index: 999;\n  ", ";\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

var PopoverComponent =
/*#__PURE__*/
function (_React$Component) {
  __chunk_1._inheritsLoose(PopoverComponent, _React$Component);

  function PopoverComponent() {
    var _this;

    for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    _this = _React$Component.call.apply(_React$Component, [this].concat(args)) || this;

    __chunk_1._defineProperty(__chunk_1._assertThisInitialized(__chunk_1._assertThisInitialized(_this)), "state", {
      placement: _this.props.placement,
      translateX: undefined,
      translateY: undefined,
      originX: undefined,
      originY: undefined
    });

    __chunk_1._defineProperty(__chunk_1._assertThisInitialized(__chunk_1._assertThisInitialized(_this)), "popper", void 0);

    __chunk_1._defineProperty(__chunk_1._assertThisInitialized(__chunk_1._assertThisInitialized(_this)), "getPopover", function () {
      return ReactDOM.findDOMNode(__chunk_1._assertThisInitialized(__chunk_1._assertThisInitialized(_this)));
    });

    __chunk_1._defineProperty(__chunk_1._assertThisInitialized(__chunk_1._assertThisInitialized(_this)), "modifier", function (data) {
      var placement = data.placement,
          offsets = data.offsets,
          arrowElement = data.arrowElement,
          arrowStyles = data.arrowStyles;
      var reference = offsets.reference,
          popper = offsets.popper;

      var _placement$split = placement.split("-"),
          position = _placement$split[0];

      var isVertical = ["top", "bottom"].indexOf(position) >= 0;
      var referenceCenter = isVertical ? reference.width / 2 : reference.height / 2;
      var side = isVertical ? "left" : "top";
      var sideValue = referenceCenter - popper[side];

      if (arrowElement) {
        var top = arrowStyles.top,
            left = arrowStyles.left;
        arrowElement.style.top = isVertical ? "" : top + "px";
        arrowElement.style.left = isVertical ? left + "px" : "";
      }

      _this.setState({
        originX: isVertical ? sideValue + "px - 50%" : 0,
        originY: !isVertical ? sideValue + "px - 50%" : 0,
        translateX: Math.round(popper.left),
        translateY: Math.round(popper.top),
        placement: data.placement
      });

      return data;
    });

    __chunk_1._defineProperty(__chunk_1._assertThisInitialized(__chunk_1._assertThisInitialized(_this)), "getOptions", function () {
      var _this$props = _this.props,
          placement = _this$props.placement,
          flip = _this$props.flip,
          shift = _this$props.shift;

      var popover = _this.getPopover();

      var arrow = popover.querySelector(s(__chunk_9.PopoverArrow));
      return {
        placement: placement,
        modifiers: {
          hide: {
            enabled: false
          },
          applyStyle: {
            enabled: false
          },
          arrow: arrow ? {
            enabled: !!arrow,
            element: arrow
          } : undefined,
          flip: {
            enabled: flip,
            padding: 16
          },
          shift: {
            enabled: shift
          },
          offset: {
            offset: "0, " + _this.props.gutter
          },
          setState: {
            order: 900,
            enabled: true,
            fn: _this.modifier
          }
        }
      };
    });

    __chunk_1._defineProperty(__chunk_1._assertThisInitialized(__chunk_1._assertThisInitialized(_this)), "initPopper", function () {
      if (!_this.popper) {
        var popover = _this.getPopover();

        var parentNode = popover.parentNode;

        if (parentNode && getComputedStyle(parentNode).display !== "none") {
          _this.popper = new Popper(parentNode, popover, _this.getOptions());
        }
      }
    });

    __chunk_1._defineProperty(__chunk_1._assertThisInitialized(__chunk_1._assertThisInitialized(_this)), "destroyPopper", function () {
      if (_this.popper) {
        _this.popper.destroy();

        _this.popper = undefined;
      }
    });

    return _this;
  }

  var _proto = PopoverComponent.prototype;

  _proto.componentDidMount = function componentDidMount() {
    this.initPopper();
  };

  _proto.componentWillUnmount = function componentWillUnmount() {
    this.destroyPopper();
  };

  _proto.componentDidUpdate = function componentDidUpdate(prevProps) {
    if (prevProps.visible !== this.props.visible || prevProps.placement !== this.props.placement || prevProps.children !== this.props.children) {
      this.destroyPopper();
      this.initPopper();
    }
  };

  _proto.getDefaultPlacement = function getDefaultPlacement() {
    var placement = this.state.placement;

    if (placement) {
      return placement.replace(/-.+$/, "");
    }

    return undefined;
  };

  _proto.render = function render() {
    var defaultPlacement = this.getDefaultPlacement();
    return React.createElement(Hidden.default, __chunk_1._extends({
      id: this.props.popoverId
    }, this.state, {
      "data-placement": this.state.placement,
      defaultSlide: defaultPlacement,
      defaultExpand: defaultPlacement,
      slideOffset: this.props.gutter
    }, this.props, {
      unmount: false
    }));
  };

  return PopoverComponent;
}(React.Component);

var Popover = styled$1.default(__chunk_4.hoist(PopoverComponent, Hidden.default))(_templateObject(), styledTools.theme("Popover")); // @ts-ignore

Popover.propTypes = {
  placement: PropTypes.oneOf(["auto", "top", "right", "bottom", "left", "top-start", "right-start", "bottom-start", "left-start", "top-end", "right-end", "bottom-end", "left-end"]),
  flip: PropTypes.bool,
  shift: PropTypes.bool,
  gutter: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
  popoverId: PropTypes.string
};
Popover.defaultProps = {
  role: "group",
  placement: "bottom",
  hideOnEsc: true,
  flip: true,
  shift: true,
  gutter: 12,
  opaque: true,
  palette: "white"
};
var Popover$1 = use$1.default(Popover, "div");

var id = -1;

var uniqueId = function uniqueId(prefix) {
  id += 1;
  return prefix ? "" + prefix + id : "" + id;
};

var onMount = function onMount(_ref) {
  var setState = _ref.setState;
  setState({
    popoverId: uniqueId("popover")
  });
};

var PopoverContainer = function PopoverContainer(props) {
  return React.createElement(__chunk_6.HiddenContainer, __chunk_1._extends({}, props, {
    onMount: __chunk_5.callAll(onMount, props.onMount)
  }));
};

function _templateObject$1() {
  var data = __chunk_1._taggedTemplateLiteralLoose(["\n  ", ";\n"]);

  _templateObject$1 = function _templateObject() {
    return data;
  };

  return data;
}

var PopoverToggleComponent = function PopoverToggleComponent(props) {
  return React.createElement(__chunk_7.HiddenToggle, __chunk_1._extends({
    "aria-expanded": props.visible,
    "aria-controls": props.popoverId,
    "aria-haspopup": true
  }, props));
};

var PopoverToggle = styled$1.default(__chunk_4.hoist(PopoverToggleComponent, __chunk_7.HiddenToggle))(_templateObject$1(), styledTools.theme("PopoverToggle")); // @ts-ignore

PopoverToggle.propTypes = {
  popoverId: PropTypes.string,
  visible: PropTypes.bool
};
var PopoverToggle$1 = use$1.default(PopoverToggle, "button");

function _templateObject$2() {
  var data = __chunk_1._taggedTemplateLiteralLoose(["\n  ", ";\n"]);

  _templateObject$2 = function _templateObject() {
    return data;
  };

  return data;
}

var PopoverShowComponent = function PopoverShowComponent(props) {
  return React.createElement(__chunk_7.HiddenShow, __chunk_1._extends({
    "aria-expanded": props.visible,
    "aria-controls": props.popoverId,
    "aria-haspopup": true
  }, props));
};

var PopoverShow = styled$1.default(__chunk_4.hoist(PopoverShowComponent, __chunk_7.HiddenShow))(_templateObject$2(), styledTools.theme("PopoverShow")); // @ts-ignore

PopoverShow.propTypes = {
  popoverId: PropTypes.string,
  visible: PropTypes.bool
};
var PopoverShow$1 = use$1.default(PopoverShow, "button");

function _templateObject$3() {
  var data = __chunk_1._taggedTemplateLiteralLoose(["\n  ", ";\n"]);

  _templateObject$3 = function _templateObject() {
    return data;
  };

  return data;
}

var PopoverHideComponent = function PopoverHideComponent(props) {
  return React.createElement(__chunk_7.HiddenHide, __chunk_1._extends({
    "aria-expanded": props.visible,
    "aria-controls": props.popoverId,
    "aria-haspopup": true
  }, props));
};

var PopoverHide = styled$1.default(__chunk_4.hoist(PopoverHideComponent, __chunk_7.HiddenHide))(_templateObject$3(), styledTools.theme("PopoverHide")); // @ts-ignore

PopoverHide.propTypes = {
  popoverId: PropTypes.string,
  visible: PropTypes.bool
};
var PopoverHide$1 = use$1.default(PopoverHide, "button");

var Popover$2 = Object.assign(Popover$1, {
  Arrow: __chunk_9.PopoverArrow,
  Container: PopoverContainer,
  Toggle: PopoverToggle$1,
  Show: PopoverShow$1,
  Hide: PopoverHide$1
});

exports.default = Popover$2;
