// reakit/lib/Provider.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

function _interopDefault (ex) { return (ex && (typeof ex === 'object') && 'default' in ex) ? ex['default'] : ex; }

var __chunk_1 = require('./chunk-4a9d70be.js');
var styled = require('styled-components');
var styled__default = _interopDefault(styled);
require('./styled.js');
var React = require('react');
var PropTypes = require('prop-types');
var constate = require('constate');

var Provider =
/*#__PURE__*/
function (_React$Component) {
  __chunk_1._inheritsLoose(Provider, _React$Component);

  function Provider() {
    return _React$Component.apply(this, arguments) || this;
  }

  var _proto = Provider.prototype;

  _proto.render = function render() {
    var _this$props = this.props,
        theme = _this$props.theme,
        children = _this$props.children,
        props = __chunk_1._objectWithoutPropertiesLoose(_this$props, ["theme", "children"]);

    return React.createElement(constate.Provider, props, theme ? React.createElement(styled.ThemeProvider, {
      theme: theme
    }, children) : children);
  };

  return Provider;
}(React.Component);

__chunk_1._defineProperty(Provider, "propTypes", {
  initialState: PropTypes.object,
  onMount: PropTypes.func,
  onUpdate: PropTypes.func,
  onUnmount: PropTypes.func,
  theme: PropTypes.oneOfType([PropTypes.object, PropTypes.func]),
  children: PropTypes.node,
  devtools: PropTypes.bool
});

__chunk_1._defineProperty(Provider, "defaultProps", {
  initialState: {}
});

exports.default = Provider;
