// reakit/lib/Sidebar.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var __chunk_1 = require('./chunk-4a9d70be.js');
var styledTools = require('styled-tools');
require('styled-components');
var styled$1 = require('./styled.js');
require('reuse');
var use$1 = require('./use.js');
var React = require('react');
var PropTypes = require('prop-types');
require('./chunk-1cbeb596.js');
require('./chunk-e7befa1f.js');
require('./Box.js');
require('@emotion/is-prop-valid');
require('react-dom');
var __chunk_4 = require('./chunk-fa7792e6.js');
require('./chunk-c40dcb9e.js');
require('./Hidden.js');
require('constate');
require('./chunk-9a6ca4f6.js');
require('./chunk-b0c0085a.js');
var Overlay = require('./Overlay.js');
var __chunk_8 = require('./chunk-a45b3f05.js');

function _templateObject() {
  var data = __chunk_1._taggedTemplateLiteralLoose(["\n  top: 0;\n  height: 100vh;\n  transform: none;\n  overflow: auto;\n  left: ", ";\n  right: ", ";\n  ", ";\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

var SidebarComponent = function SidebarComponent(props) {
  return React.createElement(Overlay.default, __chunk_1._extends({
    defaultSlide: props.align === "right" ? "left" : "right"
  }, props));
};

var Sidebar = styled$1.default(__chunk_4.hoist(SidebarComponent, Overlay.default))(_templateObject(), styledTools.ifProp({
  align: "right"
}, "auto", 0), styledTools.ifProp({
  align: "right"
}, 0, "auto"), styledTools.theme("Sidebar")); // @ts-ignore

Sidebar.propTypes = {
  align: PropTypes.oneOf(["left", "right"])
};
Sidebar.defaultProps = {
  align: "left",
  translateX: 0,
  translateY: 0
};
var Sidebar$1 = use$1.default(Sidebar, "div");

var SidebarContainer = __chunk_8.OverlayContainer;

function _templateObject$1() {
  var data = __chunk_1._taggedTemplateLiteralLoose(["\n  ", ";\n"]);

  _templateObject$1 = function _templateObject() {
    return data;
  };

  return data;
}
var SidebarToggle = styled$1.default(__chunk_8.OverlayToggle)(_templateObject$1(), styledTools.theme("SidebarToggle"));
var SidebarToggle$1 = use$1.default(SidebarToggle, "button");

function _templateObject$2() {
  var data = __chunk_1._taggedTemplateLiteralLoose(["\n  ", ";\n"]);

  _templateObject$2 = function _templateObject() {
    return data;
  };

  return data;
}
var SidebarShow = styled$1.default(__chunk_8.OverlayShow)(_templateObject$2(), styledTools.theme("SidebarShow"));
var SidebarShow$1 = use$1.default(SidebarShow, "button");

function _templateObject$3() {
  var data = __chunk_1._taggedTemplateLiteralLoose(["\n  ", ";\n"]);

  _templateObject$3 = function _templateObject() {
    return data;
  };

  return data;
}
var SidebarHide = styled$1.default(__chunk_8.OverlayHide)(_templateObject$3(), styledTools.theme("SidebarHide"));
var SidebarHide$1 = use$1.default(SidebarHide, "button");

var index = Object.assign(Sidebar$1, {
  Container: SidebarContainer,
  Toggle: SidebarToggle$1,
  Show: SidebarShow$1,
  Hide: SidebarHide$1
});

exports.default = index;
