// reakit/lib/Step.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var __chunk_1 = require('./chunk-4a9d70be.js');
var styledTools = require('styled-tools');
require('styled-components');
var styled$1 = require('./styled.js');
require('reuse');
var use$1 = require('./use.js');
var React = require('react');
var PropTypes = require('prop-types');
require('./chunk-1cbeb596.js');
require('./chunk-e7befa1f.js');
require('./Box.js');
require('@emotion/is-prop-valid');
require('react-dom');
var __chunk_4 = require('./chunk-fa7792e6.js');
require('./chunk-c40dcb9e.js');
var Hidden = require('./Hidden.js');
require('constate');
require('./chunk-9a6ca4f6.js');
var __chunk_7 = require('./chunk-b0c0085a.js');
var __chunk_10 = require('./chunk-0f6af470.js');

function _templateObject() {
  var data = __chunk_1._taggedTemplateLiteralLoose(["\n  ", ";\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

var StepComponent =
/*#__PURE__*/
function (_React$Component) {
  __chunk_1._inheritsLoose(StepComponent, _React$Component);

  function StepComponent(props) {
    var _this;

    _this = _React$Component.call(this, props) || this;
    var _this$props = _this.props,
        register = _this$props.register,
        step = _this$props.step,
        order = _this$props.order;
    register(step, order);
    return _this;
  }

  var _proto = StepComponent.prototype;

  _proto.componentDidUpdate = function componentDidUpdate(prevProps) {
    var _this$props2 = this.props,
        step = _this$props2.step,
        update = _this$props2.update,
        order = _this$props2.order;

    if (prevProps.step !== step || prevProps.order !== order) {
      update(prevProps.step, step, order);
    }
  };

  _proto.componentWillUnmount = function componentWillUnmount() {
    var _this$props3 = this.props,
        step = _this$props3.step,
        unregister = _this$props3.unregister;
    unregister(step);
  };

  _proto.render = function render() {
    var _this$props4 = this.props,
        isCurrent = _this$props4.isCurrent,
        step = _this$props4.step;
    return React.createElement(Hidden.default, __chunk_1._extends({
      unmount: true,
      visible: isCurrent(step)
    }, this.props));
  };

  return StepComponent;
}(React.Component);

var Step = styled$1.default(__chunk_4.hoist(StepComponent, Hidden.default))(_templateObject(), styledTools.theme("Step")); // @ts-ignore

Step.propTypes = {
  step: PropTypes.string.isRequired,
  register: PropTypes.func.isRequired,
  update: PropTypes.func.isRequired,
  unregister: PropTypes.func.isRequired,
  isCurrent: PropTypes.func.isRequired,
  order: PropTypes.number
};
var Step$1 = use$1.default(Step, "div");

function _templateObject$1() {
  var data = __chunk_1._taggedTemplateLiteralLoose(["\n  ", ";\n"]);

  _templateObject$1 = function _templateObject() {
    return data;
  };

  return data;
}

var toggle = function toggle(props) {
  return function () {
    return props.toggle && props.toggle(props.step);
  };
};

var StepToggleComponent = function StepToggleComponent(props) {
  return React.createElement(__chunk_7.HiddenToggle, __chunk_1._extends({}, props, {
    toggle: toggle(props)
  }));
};

var StepToggle = styled$1.default(__chunk_4.hoist(StepToggleComponent, __chunk_7.HiddenToggle))(_templateObject$1(), styledTools.theme("StepToggle")); // @ts-ignore

StepToggle.propTypes = {
  toggle: PropTypes.func.isRequired,
  step: PropTypes.string.isRequired
};
var StepToggle$1 = use$1.default(StepToggle, "button");

function _templateObject$2() {
  var data = __chunk_1._taggedTemplateLiteralLoose(["\n  ", ";\n"]);

  _templateObject$2 = function _templateObject() {
    return data;
  };

  return data;
}

var show = function show(props) {
  return function () {
    return props.show && props.show(props.step);
  };
};

var StepShowComponent = function StepShowComponent(props) {
  return React.createElement(__chunk_7.HiddenShow, __chunk_1._extends({}, props, {
    show: show(props)
  }));
};

var StepShow = styled$1.default(__chunk_4.hoist(StepShowComponent, __chunk_7.HiddenShow))(_templateObject$2(), styledTools.theme("StepShow")); // @ts-ignore

StepShow.propTypes = {
  show: PropTypes.func.isRequired,
  step: PropTypes.string.isRequired
};
var StepShow$1 = use$1.default(StepShow, "button");

function _templateObject$3() {
  var data = __chunk_1._taggedTemplateLiteralLoose(["\n  ", ";\n"]);

  _templateObject$3 = function _templateObject() {
    return data;
  };

  return data;
}
var StepHide = styled$1.default(__chunk_7.HiddenHide)(_templateObject$3(), styledTools.theme("StepHide"));
var StepHide$1 = use$1.default(StepHide, "button");

var Step$2 = Object.assign(Step$1, {
  Container: __chunk_10.StepContainer,
  Toggle: StepToggle$1,
  Show: StepShow$1,
  Hide: StepHide$1,
  Next: __chunk_10.StepNext,
  Previous: __chunk_10.StepPrevious
});

exports.default = Step$2;
