// reakit/lib/Tooltip.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var __chunk_1 = require('./chunk-4a9d70be.js');
var styledTools = require('styled-tools');
require('styled-components');
var styled$1 = require('./styled.js');
require('reuse');
var use$1 = require('./use.js');
var React = require('react');
require('prop-types');
require('./chunk-1cbeb596.js');
require('./chunk-e7befa1f.js');
require('./Box.js');
require('@emotion/is-prop-valid');
var ReactDOM = require('react-dom');
var __chunk_4 = require('./chunk-fa7792e6.js');
require('./chunk-c40dcb9e.js');
require('./Hidden.js');
require('constate');
require('./chunk-9a6ca4f6.js');
require('./chunk-b0c0085a.js');
require('styled-selector');
require('popper.js');
var __chunk_9 = require('./chunk-dbf1c71e.js');
var Popover = require('./Popover.js');

function _templateObject() {
  var data = __chunk_1._taggedTemplateLiteralLoose(["\n  ", ";\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

var TooltipComponent =
/*#__PURE__*/
function (_React$Component) {
  __chunk_1._inheritsLoose(TooltipComponent, _React$Component);

  function TooltipComponent() {
    var _this;

    for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    _this = _React$Component.call.apply(_React$Component, [this].concat(args)) || this;

    __chunk_1._defineProperty(__chunk_1._assertThisInitialized(__chunk_1._assertThisInitialized(_this)), "state", {
      visible: false
    });

    __chunk_1._defineProperty(__chunk_1._assertThisInitialized(__chunk_1._assertThisInitialized(_this)), "isControlled", function () {
      return typeof _this.props.visible !== "undefined";
    });

    __chunk_1._defineProperty(__chunk_1._assertThisInitialized(__chunk_1._assertThisInitialized(_this)), "show", function () {
      return _this.setState({
        visible: true
      });
    });

    __chunk_1._defineProperty(__chunk_1._assertThisInitialized(__chunk_1._assertThisInitialized(_this)), "hide", function () {
      return _this.setState({
        visible: false
      });
    });

    __chunk_1._defineProperty(__chunk_1._assertThisInitialized(__chunk_1._assertThisInitialized(_this)), "getTooltip", function () {
      return ReactDOM.findDOMNode(__chunk_1._assertThisInitialized(__chunk_1._assertThisInitialized(_this)));
    });

    return _this;
  }

  var _proto = TooltipComponent.prototype;

  _proto.componentDidMount = function componentDidMount() {
    var _this$getTooltip = this.getTooltip(),
        parentNode = _this$getTooltip.parentNode;

    if (parentNode && !this.isControlled()) {
      parentNode.addEventListener("mouseenter", this.show);
      parentNode.addEventListener("focus", this.show);
      parentNode.addEventListener("mouseleave", this.hide);
      parentNode.addEventListener("blur", this.hide);
    }
  };

  _proto.componentWillUnmount = function componentWillUnmount() {
    var _this$getTooltip2 = this.getTooltip(),
        parentNode = _this$getTooltip2.parentNode;

    if (parentNode) {
      parentNode.removeEventListener("mouseenter", this.show);
      parentNode.removeEventListener("focus", this.show);
      parentNode.removeEventListener("mouseleave", this.hide);
      parentNode.removeEventListener("blur", this.hide);
    }
  };

  _proto.render = function render() {
    var _ref = this.isControlled() ? this.props : this.state,
        visible = _ref.visible;

    return React.createElement(Popover.default, __chunk_1._extends({}, this.props, {
      visible: visible
    }));
  };

  return TooltipComponent;
}(React.Component);

var Tooltip = styled$1.default(__chunk_4.hoist(TooltipComponent, Popover.default))(_templateObject(), styledTools.theme("Tooltip"));
Tooltip.defaultProps = {
  role: "tooltip",
  placement: "top",
  opaque: true,
  palette: "grayscale"
};
var Tooltip$1 = use$1.default(Tooltip, "div");

function _templateObject$1() {
  var data = __chunk_1._taggedTemplateLiteralLoose(["\n  ", ";\n"]);

  _templateObject$1 = function _templateObject() {
    return data;
  };

  return data;
}
var TooltipArrow = styled$1.default(__chunk_9.PopoverArrow)(_templateObject$1(), styledTools.theme("TooltipArrow"));
TooltipArrow.defaultProps = {
  palette: "grayscale"
};
var TooltipArrow$1 = use$1.default(TooltipArrow, "div");

var index = Object.assign(Tooltip$1, {
  Arrow: TooltipArrow$1
});

exports.default = index;
